annex/src/example9.cpp

```diff
@@ -0,0 +1,62 @@
+/****************************************************************************
+*	1 event loop
+*	Listener to print the content of the event and the thread id
+*	Publish and subscriber for other types of data: int, my own strct, etc.
+*	Number as the output
+*****************************************************************************/
+
+#include <iostream>
+
+#include <klepsydra/high_performance/event_loop_middleware_provider.h>
+
+#include <thread>
+
+#include <chrono>
+
+# include "int_publisher.h"
+
+# include "position_publisher.h"
+
+int main() {
+	kpsr::high_performance::EventLoopMiddlewareProvider<16> eventloop(nullptr);
+
+	eventloop.start();
+	// main thread id to print here
+	eventloop.getSubscriber<int>("temperature receiver")->registerListener("listener", [](const int & message) {
+		std::cout <<"temperature received t : " << message << std::endl;
+		std::thread::id this_id = std::this_thread::get_id();
+		std::cout << "the temperatureListener id is : " << this_id << std::endl;
+	});
+
+	eventloop.getSubscriber<Position>("position receiver")->registerListener("listener", [](const Position & message) {
+		std::cout <<"position received x : " << message.x << " y : " << message.y << std::endl;
+		std::thread::id this_id = std::this_thread::get_id();
+		std::cout << "the positionListener id is : " << this_id << std::endl;
+	});
+
+	IntPublisher intPublisher(eventloop.getPublisher<int>("temperature receiver", 0, nullptr, nullptr));
+	intPublisher.run();
+	
+	PositionPublisher positionPublisher(eventloop.getPublisher<Position>("position receiver", 0, nullptr, nullptr));
+	std::thread t2([&positionPublisher, &intPublisher]() {
+	// print the thread id here as well
```

Comment:
You have added this comment but not completed the task it says.

Reply:
Done